Source/Particles/Deposition/ChargeDeposition.H

```diff
@@ -205,19 +250,19 @@ void doChargeDepositionShapeN (const GetParticlePosition& GetPosition,
                 k = compute_shape_factor(sz, z - 0.5_rt);
             }
 
-            // Deposit charge into rho_tmp
+            // Deposit charge into buf
 #if (defined WARPX_DIM_XZ) || (defined WARPX_DIM_RZ)
             for (int iz=0; iz<=depos_order; iz++){
                 for (int ix=0; ix<=depos_order; ix++){
                     amrex::Gpu::Atomic::AddNoRet(
-                        &rho_tmp(lo.x+i+ix, lo.y+k+iz, 0, 0),
+                        &buf(lo.x+i+ix, lo.y+k+iz, 0, 0),
```

Comment:
I'm getting out of bounds accesses here, by up to a few points, positive or negative, in both X and Z. E.g.

` (-10,5,0,0) is out of bound (-2:130,-2:10,0:0,0:0)`
` (29,8,0,0) is out of bound (27:32,-2:3,0:0,0:0)`

There are hundreds of these in the first call to `depositChargeShapeN`. The logic all looks reasonable to me, but there must be some subtlety in the box sizing calculations that I'm missing.